math/mathcore/CMakeLists.txt

```
@@ -106,8 +106,12 @@ if(runtime_cxxmodules)
   # FIXME: Revise after a llvm upgrade or reproduce it outside rootcling.
   list(REMOVE_ITEM HEADERS "Math/Math.h")
 
-  if(veccore)
-    set(dictoptions "-mByproduct" "Vc")
+  if(veccore OR vc)
+    # We do not link against libVc.a thus it makes no sense to check for
+    # version compatibility between libraries and header files. This fixes
+    # ROOT-11002 where upon building the modules.idx we run the static ctor
```

Why does building the module index *run* anything?

---

Because when building the modules.idx we need to load all files. That also mean doing static initialization for the objects as the usual rules apply.

---

rootcling usually has syntax-only mode, i.e. no execution engine. What am I missing?

---

@vgvassilev ping

---

I do not understand your question. Rootcling generates a module from headers which have static initializers. When loading that module by root we run the initializers. This means that the way we build the .so file and the module needs to be the same. We do not find the Vc versioning symbol anywhere (it is not in MathCore) meaning we need to just build the module in the same way.

---

My understanding was that the error message is emitted by `rootcling` (likely from the comment "upon building the modules.idx we run the static ctor"), not `root`. Your latest comment here clarifies it, but to avoid my confusion maybe reword:
```suggestion
    # ROOT-11002 where modules.idx triggers running the Vc's static ctor
```

---

The modules.idx is built byte root.exe.